packages/astro/src/assets/vite-plugin-assets.ts

```diff
@@ -213,6 +214,12 @@ export default function assets({ settings }: { settings: AstroSettings }): vite.
 						});
 					}
 
+					if (settings.config.experimental.svg && /\.svg$/.test(id)) {
+						const { contents, ...metadata } = imageMetadata;
+						// We know that the contents are present, as we only emit this property for SVG files
+						return makeSvgComponent(metadata, contents!, { mode: settings.config.experimental.svg.mode });
+					}
```

Comment:
Is the intent of the if condition that `settings.config.experimental.svg` must be set, e.g. `{}`, to opt in to this feature? Or is the feature will always be enabled?

Reply:
yeah, this wasn't quite sure how to handle this. Wanted it to be easy to opt into but wanted options as well. so i wanted a `boolean | { mode: 'inline' | 'sprite' }` but wasn't sure how to handle that.

Reply:
That types looks good to me. You can declare it like this: https://github.com/stramel/astro/blob/fcc6eca76e427e50f39263ebdd579bdecb1f573c/packages/astro/src/core/config/schema.ts#L238-L246

And handle that the types could be a boolean or object. You can normalized it to an object with `.transform()` in the schema if you'd like.